packages/stream-tests/src/__tests__/basic-indexing.test.ts

```diff
@@ -0,0 +1,252 @@
+    })
+    expect(resultObj1.pageInfo.hasNextPage).toBeTruthy()
+    const resultObj2 = await ceramic.index.queryIndex({
+      model: model.id,
+      first: 2,
+      after: resultObj1.pageInfo.endCursor,
+    })
+    expect(resultObj2.pageInfo.hasNextPage).toBeFalsy()
+
+    const results = (
+      await Promise.all([
+        extractDocuments(ceramic, resultObj0),
+        extractDocuments(ceramic, resultObj1),
+        extractDocuments(ceramic, resultObj2),
+      ])
+    ).flat()
+
+    expect(results.length).toEqual(5)
+    expect(results[0].id.toString()).toEqual(doc1.id.toString())
+    expect(results[0].content).toEqual(CONTENT1)
+    expect(results[1].id.toString()).toEqual(doc2.id.toString())
+    expect(results[1].content).toEqual(CONTENT2)
+    expect(results[2].id.toString()).toEqual(doc3.id.toString())
+    expect(results[2].content).toEqual(CONTENT3)
+    expect(results[3].id.toString()).toEqual(doc4.id.toString())
+    expect(results[3].content).toEqual(CONTENT4)
+    expect(results[4].id.toString()).toEqual(doc5.id.toString())
+    expect(results[4].content).toEqual(CONTENT5)
+  })
+
+  test('multiple documents - one page, reverse order', async () => {
+    const doc1 = await ModelInstanceDocument.create(ceramic, CONTENT0, midMetadata)
+    await doc1.replace(CONTENT1)
+    const doc2 = await ModelInstanceDocument.create(ceramic, CONTENT2, midMetadata)
+    const doc3 = await ModelInstanceDocument.create(ceramic, CONTENT3, midMetadata)
+
+    console.log(`docIds: [${doc1.id.toString()}, ${doc2.id.toString()}, ${doc3.id.toString()}]`)
+
+    const resultObj = await ceramic.index.queryIndex({ model: model.id, last: 100 })
+    const results = await extractDocuments(ceramic, resultObj)
+
+    expect(results.length).toEqual(3)
+    expect(results[0].id.toString()).toEqual(doc3.id.toString())
+    expect(results[0].content).toEqual(CONTENT3)
+    expect(results[1].id.toString()).toEqual(doc2.id.toString())
+    expect(results[1].content).toEqual(CONTENT2)
+    expect(results[2].id.toString()).toEqual(doc1.id.toString())
+    expect(results[2].content).toEqual(CONTENT1)
+  })
+
+  test('multiple documents - multiple pages, reverse order', async () => {
+    const doc1 = await ModelInstanceDocument.create(ceramic, CONTENT0, midMetadata)
+    await doc1.replace(CONTENT1)
+    const doc2 = await ModelInstanceDocument.create(ceramic, CONTENT2, midMetadata)
+    const doc3 = await ModelInstanceDocument.create(ceramic, CONTENT3, midMetadata)
+    const doc4 = await ModelInstanceDocument.create(ceramic, CONTENT4, midMetadata)
+    const doc5 = await ModelInstanceDocument.create(ceramic, CONTENT5, midMetadata)
+
+    const resultObj0 = await ceramic.index.queryIndex({ model: model.id, last: 2 })
+    expect(resultObj0.pageInfo.hasPreviousPage).toBeTruthy()
+    const resultObj1 = await ceramic.index.queryIndex({
+      model: model.id,
+      last: 2,
+      before: resultObj0.pageInfo.endCursor,
```

Comment:
Should be before the `startCursor` here I think, otherwise you're querying the same items from the end, except off by one.